### PR TITLE
AOCC support for WRFv3.9.1.1

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
Added AOCC Support for 
1. HDF5
2. NetCDF-Fortran
3. WRF 3.9.1.1

### CODE REVIEW

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 1]
```diff
@@ -324,6 +324,15 @@ def patch_postdeps(self):
                     arg for arg in m.group(1).split(' ') if arg != '-l'),
                 'libtool')
 
+    @run_after('configure')
+    def patch_libtool(self):
+        """AOCC support for HDF5"""
+        if '%aocc' in self.spec:
+            filter_file(
+                '\\$wl-soname \\$wl\\$soname',
+                '-fuse-ld=ld -Wl,-soname,\\$soname',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hdf5/package.py]:
```suggestion
                r'\$wl-soname \$wl\$soname',
                r'-fuse-ld=ld -Wl,-soname,\$soname',
```

File: var/spack/repos/builtin/packages/netcdf-fortran/package.py [hunk 2]
```diff
@@ -136,6 +136,15 @@ def configure_args(self):
 
         return config_args
 
+    @run_after('configure')
+    def patch_libtool(self):
+        """AOCC support for NETCDF-F"""
+        if '%aocc' in self.spec:
+            filter_file(
+                '\\${wl}-soname \\$wl\\$soname',
+                '-fuse-ld=ld -Wl,-soname,\\$soname',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/netcdf-fortran/package.py]:
```suggestion
                r'\${wl}-soname \$wl\$soname',
                r'-fuse-ld=ld -Wl,-soname,\$soname',
```

**Review by adamjstewart — COMMENTED** [review]
@MichaelLaufer  @WardF  @lrknox  @ptooley  @skosukhin can you review?

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 1 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/hdf5/package.py]: Addressed above suggestion

File: var/spack/repos/builtin/packages/netcdf-fortran/package.py (hunk 2 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/netcdf-fortran/package.py]: Addressed above suggestion

**Comment by MichaelLaufer** [comment]
@adamjstewart @amd-toolchain-support 
WRF changes look fine to me.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**